mongodb/vibe/db/mongo/collection.d

```diff
@@ -493,6 +497,46 @@ struct MongoCollection {
 		enforce(reply["ok"].get!double == 1, "dropIndex command failed: "~reply["errmsg"].opt!string);
 	}
 
+	/**
+		Creates indexes on the collection.
+	*/
+	void createIndex(T)(T query) 
+	@safe {
+		static struct Indexes {
+			T key;
+		}
+
+		static struct CMD {
+			string createIndexes;
+			Indexes indexes;
+		}
+
+		CMD cmd;
+		cmd.createIndexes = m_name;
+		cmd.indexes.key = query;
+		auto reply = database.runCommand(cmd);
+		enforce(reply["ok"].get!double == 1, "createIndex command failed: "~reply["errmsg"].opt!string);
+	}
+
+	/**
+		Returns an array that holds a list of documents that identify and describe the existing indexes on the collection. 
+	*/
+	R getIndexes(T = Bson, R = Bson)() 
+	@safe {
+		static struct CMD {
+			string listIndexes;
+		}
+
+		CMD cmd;
+		listIndexes = m_name;
+
+		auto reply = database.runCommand(cmd);
+		return reply;
```

Comment:
I think this returns the raw protocol response, it should probably return the data value it returns instead? The raw response Bson is probably not interesting for the user if they use a high level method for this.

https://docs.mongodb.com/manual/reference/command/listIndexes/#listIndexes.cursor

The docs say this returns a cursor so the return type of the function should be a cursor too.